### PR TITLE
Fix the bug that layers got updated in the middle of the picking cycle

### PR DESCRIPTION
**Pull request by shaojingli**
@Pessimistress 
please take a look and see if I understand your request correctly

@ibgreen

### CODE REVIEW

File: src/lib/draw-and-pick.js [hunk 1]
```diff
@@ -152,6 +155,12 @@ export function pickLayers(gl, {
         layer = layer.parentLayer;
       }
 
+      // This guarantees that there will be only one copy of info with
+      // the same composite layer
+      infos.set(info.layer.id, info);
```
Comment (Pessimistress) [inline src/lib/draw-and-pick.js]: Check if `info` is empty

**Comment by shaojingli** [comment]
@Pessimistress

**Comment by shaojingli** [comment]
#443

File: src/lib/draw-and-pick.js [hunk 2]
```diff
@@ -143,21 +150,38 @@ export function pickLayers(gl, {
...
+      // This guarantees that there will be only one copy of info for
+      // one composite layer
+      if (info) {
+        infos.set(info.layer.id, info);
+      }
+    });
+
+    infos.forEach(info => {
       // If layer.getPickingInfo() returns null, do not proceed
       if (info) {
```
Comment (Pessimistress) [inline src/lib/draw-and-pick.js]: Not necessary since only non-empty info objects were added to the map

**Pull request merged**